plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/stock-indicator/block.tsx

```diff
@@ -18,15 +21,54 @@ import type { BlockAttributes } from './types';
 
 type Props = BlockAttributes & HTMLAttributes< HTMLDivElement >;
 
+/**
+ * Determines whether the stock indicator should be visible based on product type and availability.
+ *
+ * @param product                             The product.
+ * @param availabilityText                    The stock availability text.
+ * @param isDescendentOfSingleProductTemplate Whether this is descendent of single product template.
+ * @param selectedProductType                 The selected product type.
+ * @return True if stock indicator should be visible.
+ */
+const isStockIndicatorVisible = (
+	product: ProductResponseItem,
+	availabilityText: string,
+	isDescendentOfSingleProductTemplate: boolean | undefined,
+	selectedProductType: string | undefined
+) => {
+	const productTypesWithoutStockIndicator = getSetting< string[] >(
+		'productTypesWithoutStockIndicator',
+		[ 'external', 'grouped', 'variable' ]
+	);
+
+	const productType = selectedProductType || product?.type;
+
+	const isProductTypeAllowed =
+		! productTypesWithoutStockIndicator.includes( productType );
+
+	return (
+		( isProductTypeAllowed && availabilityText !== '' ) ||
+		isDescendentOfSingleProductTemplate
```

Comment:
Why do we want to make it always visible if it's a descendant of the Single Product template? Shouldn't the visibility of the block depend on the product type? :thinking:

Reply:
Good call! The condition was wrong. I fixed it in the commit [82c57c5](https://github.com/woocommerce/woocommerce/pull/53358/commits/82c57c566e23bffa8935a26aedd148f64f195bf8)